Lib/asyncio/base_events.py

```diff
@@ -333,6 +366,11 @@ def sockets(self):
         return tuple(trsock.TransportSocket(s) for s in self._sockets)
 
     def close(self):
+        if self._state == _ServerState.CLOSED or self._state == _ServerState.SHUTDOWN:
```

Comment:
I typically follow this idiom, but it doesn't matter that much:
```suggestion
        if self._state in {_ServerState.CLOSED, _ServerState.SHUTDOWN}:
```